tsfile/src/main/java/org/apache/iotdb/tsfile/read/query/executor/TsFileExecutor.java

```diff
@@ -102,18 +114,23 @@ private QueryDataSet execute(List<Path> selectedPathList) throws IOException {
    * @return DataSet without TimeGenerator
    */
   private QueryDataSet execute(List<Path> selectedPathList, GlobalTimeExpression timeFilter)
-      throws IOException {
+      throws IOException, NoMeasurementException {
     List<FileSeriesReader> readersOfSelectedSeries = new ArrayList<>();
     List<TSDataType> dataTypes = new ArrayList<>();
 
     for (Path path : selectedPathList) {
       List<ChunkMetaData> chunkMetaDataList = metadataQuerier.getChunkMetaDataList(path);
-      FileSeriesReader seriesReader = new FileSeriesReaderWithFilter(chunkLoader, chunkMetaDataList,
-          timeFilter.getFilter());
+      FileSeriesReader seriesReader;
+      if (chunkMetaDataList.isEmpty()) {
+        seriesReader = new EmptyFileSeriesReader();
+        dataTypes.add(metadataQuerier.getDataType(path.getMeasurement()));
+      } else {
+        seriesReader = new FileSeriesReaderWithFilter(chunkLoader, chunkMetaDataList,
+            timeFilter.getFilter());
+        dataTypes.add(chunkMetaDataList.get(0).getTsDataType());
+      }
       readersOfSelectedSeries.add(seriesReader);
-      dataTypes.add(chunkMetaDataList.get(0).getTsDataType());
     }
```

Comment:
The two code blocks(this and the one above) look exactly the same. Maybe you can extract a function from them?

Reply:
fixed.